lib/internal/process/pre_execution.js

```diff
@@ -513,6 +518,48 @@ function initializeClusterIPC() {
   }
 }
 
+function initializePermission() {
+  const experimentalPermission = getOptionValue('--experimental-permission');
+  if (experimentalPermission) {
+    process.emitWarning('Permission is an experimental feature',
+                        'ExperimentalWarning');
+    const { has, deny } = require('internal/process/permission');
+    const warnFlags = [
+      '--allow-child-process',
+      '--allow-worker',
+    ];
+    for (const flag of warnFlags) {
+      if (getOptionValue(flag)) {
+        process.emitWarning(
+          `The flag ${flag} must be used with extreme caution. ` +
+        'It could invalidate the permission model.', 'SecurityWarning');
+      }
+    }
+
+    ObjectDefineProperty(process, 'permission', {
+      __proto__: null,
+      enumerable: true,
+      configurable: false,
+      value: {
+        has,
+        deny,
+      }
+    });
+  } else {
+    const availablePermissionFlags = [
+      '--allow-fs-read',
+      '--allow-fs-write',
+      '--allow-child-process',
+      '--allow-worker',
+    ];
+    for (const flag of availablePermissionFlags) {
+      if (getOptionValue(flag)) {
+        throw new ERR_MISSING_OPTION('--experimental-permission');
+      }
+    }
+  }
```

Comment:
Don't we have a C++ util for this? @RaisinTen

Reply:
@aduh95 unfortunately, I don't think there's one to enforce that a different CLI option should be passed to enable another CLI option. The only somewhat complex operation the CLI parser supports is aliases that can be expanded.